Author: Jan Christoph Ebersbach <jceb@e-jc.de>
URL: no URL yet
Extension to the toggleview function.  The first client from the added/toggled
tag is focussed.  This emulates a simple scratch pad functionality by keeping
the "scratch pad" window(s) on a certain tag that can be toggled by pressing an
easy to reach keybinding.

Index: dwm/config.def.h
===================================================================
--- dwm/config.def.h.orig
+++ dwm/config.def.h
@@ -70,6 +70,7 @@ static char dmenumon[2] = "0"; /* compon
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbgcolor, "-sf", selfgcolor, NULL };
 static const char *termcmd[]  = { "x-terminal-emulator", NULL };
 
+#include "toggleview_focus.c"
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
@@ -87,6 +88,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
+	{ MODKEY,                       XK_space,  toggleview_focus, {.ui = 1} },
 	{ MODKEY|ControlMask,           XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
Index: dwm/toggleview_focus.c
===================================================================
--- /dev/null
+++ dwm/toggleview_focus.c
@@ -0,0 +1,13 @@
+static void toggleview_focus(const Arg *arg) {
+	Client *c;
+	toggleview(arg);
+	restack(selmon);
+	for(c = selmon->clients; c; c = c->next) {
+		if (c->tags & arg->ui && c->isfloating) {
+			XRaiseWindow(dpy, c->win);
+		}
+	}
+	for(c = selmon->clients; !(c->tags & arg->ui) && c->next; c = c->next) ;
+	if(c && c->tags & arg->ui)
+		focus(c);
+}
